File: test/conformance/host-fixtures/bytes_roundtrip.js
let calls = 0;
module.exports = {
  echoBytes(b) { calls += 1; return b; },
  nullableBytes(b) { calls += 1; return b; },
  makeBytes(n) { calls += 1; return new Uint8Array(n); },
  readByte(b) { calls += 1; return b[0]; },
  callCount() { return calls; },
  badBytesReturn() { calls += 1; return "not-bytes"; }
};

File: test/conformance/host-fixtures/bytes_roundtrip.java
public final class HostBytes_roundtrip {
  private static int calls;
  private static $DealRt.Bytes shared = new $DealRt.Bytes(new byte[2]);
  public static Object echoBytes($DealRt.Bytes b) { calls += 1; return b; }
  public static Object nullableBytes($DealRt.Bytes b) { calls += 1; return b; }
  public static Object makeBytes(int n) { calls += 1; return new $DealRt.Bytes(new byte[n]); }
  public static Object sharedBytes() { return shared; }
  public static int readByte($DealRt.Bytes b) { calls += 1; return b.data[0] & 0xFF; }
  public static int callCount() { return calls; }
  public static Object badBytesReturn() { calls += 1; return "not-bytes"; }
}
